matplotcheck/raster.py

```diff
@@ -140,25 +140,36 @@ def assert_legend_accuracy_classified_image(
         # Check that each legend entry label is in one of all_label_options
         assert len([val for val in legend_dict.values() if val]) == len(
             all_label_options
-        ), "Incorrect legend labels"
+        ), (
+            "Number of label options provided doesn't match the number of",
+            " labels found in the image.",
+        )
 
         # Create two copies of image array, one filled with the plot data class
         # labels (im_data_labels) and the other with the expected labels
         # (im_expected_labels)
         im_class_dict = {}
+
+        # Accounting for images that don't start at 0
+        correction = np.min(im_data)
+
         for val in np.unique(im_data):
-            im_class_dict[val] = legend_dict[im_cmap(im.norm(val))]
+            im_class_dict[val - correction] = legend_dict[
+                im_cmap(im.norm(val))
+            ]
         im_data_labels = [
-            [im_class_dict[val] for val in row] for row in im_data.data
+            [im_class_dict[val - correction] for val in row]
+            for row in im_data.data
         ]
         im_expected_labels = [
-            [all_label_options[val][0] for val in row] for row in im_expected
+            [all_label_options[val - correction][0] for val in row]
```

Comment:
just a note - this method is super long and i think the parts of it could be modulearized and some reused elsewhere (get_image) for example. so let's clean this up.

I see that the notebook is working but i think we should clean up this method and then look at how is works more closely but looking at the various methods! 

note: CI is failing now but it's an known issue and a new one so we can keep working and assume it will fail. if i need to remove m2r we can consider that... i left a comment about m2r and the failure in my most recent pr to update docs!